warehouse/query-core/src/main/java/datawave/query/jexl/nodes/ExceededOrThresholdMarkerJexlNode.java

```diff
@@ -45,65 +49,35 @@ public ExceededOrThresholdMarkerJexlNode(JexlNode node) {
         super(node);
     }
     
-    public ExceededOrThresholdMarkerJexlNode(String fieldname, URI fstPath) {
-        // Create an assignment for the field name and fst paths
-        JexlNode fieldAssignment = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(FIELD_PROP, fieldname));
-        JexlNode fstAssignment = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(FST_URI_PROP, fstPath.toString()));
+    public ExceededOrThresholdMarkerJexlNode(String fieldname, URI fstPath) throws JsonProcessingException {
         
-        // wrap the assignments in an AND node
-        JexlNode andNode = JexlNodeFactory.createUnwrappedAndNode(Arrays.asList(fieldAssignment, fstAssignment));
+        ExceededOrParams params = new ExceededOrParams(fieldname, fstPath.toString());
+        
+        // Create an assignment for the params
+        JexlNode paramsNode = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(EXCEEDED_OR_PARAMS, objectMapper.writeValueAsString(params)));
         
         // now set the source
-        setupSource(andNode);
+        setupSource(paramsNode);
     }
     
-    public ExceededOrThresholdMarkerJexlNode(String fieldname, Collection<String> values) {
-        // Create an assignment for the field name
-        JexlNode fieldAssignment = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(FIELD_PROP, fieldname));
-        
-        // Create an assignment for the values
-        JexlNode valuesAssignment = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(VALUES_PROP,
-                        StringUtils.join(StringUtils.COMMA_STR, new EscapedCollection(values))));
+    public ExceededOrThresholdMarkerJexlNode(String fieldname, Collection<String> values) throws JsonProcessingException {
+        ExceededOrParams params = new ExceededOrParams(fieldname, values);
         
-        // wrap the assignments in an AND node
-        JexlNode andNode = JexlNodeFactory.createUnwrappedAndNode(Arrays.asList(fieldAssignment, valuesAssignment));
+        // Create an assignment for the params
+        JexlNode paramsNode = JexlNodeFactory.createExpression(JexlNodeFactory.createAssignment(EXCEEDED_OR_PARAMS, objectMapper.writeValueAsString(params)));
```

Comment:
All of the constructors here construct an ExceededOrParams and then create the jexlNode and invoke setupSource. We could make one constructor that accepts an ExceededOrParams object, creates the paramsNode, and calls setupSource. Then these constructors could all create the ExceededOrParams and delegate to it. That'd eliminate some duplicated logic.

Reply:
I kind of wanted to keep the ExceededOrParams class internal to the ExceededOrThresholdMarkerJexlNode to prevent the creation of ExceededOrParams with invalid combinations (i.e. fstPath & values, or fstPath & ranges).  Let me know how you feel about the factory methods I settled on in the latest commit.  I also made the ExceededOrParams class private.